UI/window-dock-youtube-app.cpp

```diff
@@ -0,0 +1,383 @@
+#include <QUuid>
+
+#include "window-basic-main.hpp"
+#include "youtube-api-wrappers.hpp"
+#include "window-dock-youtube-app.hpp"
+
+#include "ui-config.h"
+
+#ifndef YOUTUBE_WEBAPP_PLACEHOLDER
+#define YOUTUBE_WEBAPP_PLACEHOLDER \
+	"https://studio.youtube.com/live/channel/UC/console?kc=OBS"
+#endif
+#ifndef YOUTUBE_WEBAPP_ADDRESS
+#define YOUTUBE_WEBAPP_ADDRESS \
+	"https://studio.youtube.com/live/channel/%1/console?kc=OBS"
+#endif
+
+#define BROADCAST_CREATED "BROADCAST_CREATED"
+#define BROADCAST_SELECTED "BROADCAST_SELECTED"
+#define INGESTION_STARTED "INGESTION_STARTED"
+#define INGESTION_STOPPED "INGESTION_STOPPED"
+#define LOGGED_OUT "LOGGED_OUT"
+
+YouTubeAppDock::YouTubeAppDock()
+	: BrowserDock(), dockBrowser(nullptr), cookieManager(nullptr)
+{
+	OBSBasic::InitBrowserPanelSafeBlock();
+	AddYouTubeAppDock("YouTube Live Control Panel");
+}
+
+YouTubeAppDock::~YouTubeAppDock()
+{
+	if (cookieManager) {
+		cookieManager->FlushStore();
+		delete cookieManager;
+	}
+}
+
+bool YouTubeAppDock::IsYTServiceSelected()
+{
+	obs_service_t *service_obj = OBSBasic::Get()->GetService();
+	OBSDataAutoRelease settings = obs_service_get_settings(service_obj);
+	const char *service = obs_data_get_string(settings, "service");
+	return IsYouTubeService(service);
+}
+
+void YouTubeAppDock::AccountConnected()
+{
+	channelId.clear(); // renew channel id
+	UpdateChannelId();
+}
+
+void YouTubeAppDock::AccountDisconnected()
+{
+	SettingsUpdated(true);
+}
+
+void YouTubeAppDock::SettingsUpdated(bool cleanup)
+{
+	bool ytservice = IsYTServiceSelected();
+	SetVisibleYTAppDockInMenu(ytservice);
+
+	// definitely cleanup if YT switched off
+	if (!ytservice || cleanup) {
+		if (cookieManager)
+			cookieManager->DeleteCookies("", "");
+	}
+	if (ytservice)
+		Update();
+}
+
+std::string YouTubeAppDock::InitYTUserUrl()
+{
+	std::string user_url(YOUTUBE_WEBAPP_PLACEHOLDER);
+
+	if (IsUserSignedIntoYT()) {
+		YoutubeApiWrappers *apiYouTube = GetYTApi();
+		if (apiYouTube) {
+			ChannelDescription channel_description;
+			if (apiYouTube->GetChannelDescription(
+				    channel_description)) {
+				QString url =
+					QString(YOUTUBE_WEBAPP_ADDRESS)
+						.arg(channel_description.id);
+				user_url = url.toStdString();
+			} else {
+				blog(LOG_ERROR,
+				     "YT: InitYTUserUrl() Failed to get channel id");
+			}
+		}
+	} else
+		blog(LOG_ERROR, "YT: InitYTUserUrl() User is not signed");
```

Comment:
If you enclose one branch of an `if` statement, then all branches must be enclosed.

Reply:
Done.